src/Microsoft.Identity.Web/MicrosoftIdentityCircuitHandler.cs

```diff
@@ -17,257 +17,257 @@
 
-        /// <summary>
-        /// Initializes a new instance of the <see cref="MicrosoftIdentityConsentAndConditionalAccessHandler"/> class.
-        /// </summary>
-        /// <param name="serviceProvider">Service provider to get the HttpContextAccessor for the current HttpContext, when available.</param>
-        public MicrosoftIdentityConsentAndConditionalAccessHandler(IServiceProvider serviceProvider)
-        {
-            _httpContextAccessor = serviceProvider.GetService<IHttpContextAccessor>();
-        }
+		/// <summary>
+		/// Initializes a new instance of the <see cref="MicrosoftIdentityConsentAndConditionalAccessHandler"/> class.
+		/// </summary>
+		/// <param name="serviceProvider">Service provider to get the HttpContextAccessor for the current HttpContext, when available.</param>
+		public MicrosoftIdentityConsentAndConditionalAccessHandler(IServiceProvider serviceProvider)
+		{
+			_httpContextAccessor = serviceProvider.GetService<IHttpContextAccessor>();
+		}
 
-        /// <summary>
-        /// Boolean to determine if server is Blazor.
-        /// </summary>
-        public bool IsBlazorServer { get; set; }
+		/// <summary>
+		/// Boolean to determine if server is Blazor.
+		/// </summary>
+		public bool IsBlazorServer { get; set; }
 
-        /// <summary>
-        /// Current user.
-        /// </summary>
-        public ClaimsPrincipal User
-        {
-            get
-            {
-                return _user ??
+		/// <summary>
+		/// Current user.
+		/// </summary>
+		public ClaimsPrincipal User
+		{
+			get
+			{
+				return _user ??
 #pragma warning disable CS8602 // Dereference of a possibly null reference. HttpContext will not be null in this case.
-                    (!IsBlazorServer ? _httpContextAccessor.HttpContext.User :
+					(!IsBlazorServer ? _httpContextAccessor.HttpContext.User :
 #pragma warning restore CS8602 // Dereference of a possibly null reference. HttpContext will not be null in this case.
-                    throw new InvalidOperationException(IDWebErrorMessage.BlazorServerUserNotSet));
-            }
-            set
-            {
-                _user = value;
-            }
-        }
+					throw new InvalidOperationException(IDWebErrorMessage.BlazorServerUserNotSet));
+			}
+			set
+			{
+				_user = value;
+			}
+		}
 
-        /// <summary>
-        /// Base URI to use in forming the redirect.
-        /// </summary>
-        public string? BaseUri
-        {
-            get
-            {
-                return _baseUri ??
+		/// <summary>
+		/// Base URI to use in forming the redirect.
+		/// </summary>
+		public string? BaseUri
+		{
+			get
+			{
+				return _baseUri ??
 #pragma warning disable CS8602 // Dereference of a possibly null reference. HttpContext will not be null in this case
-                    (!IsBlazorServer ? CreateBaseUri(_httpContextAccessor.HttpContext.Request) :
+					(!IsBlazorServer ? CreateBaseUri(_httpContextAccessor.HttpContext.Request) :
```

Comment:
Everytime HttpContext is touched, it will need somesort of lock in this code.

Reply:
https://github.com/AzureAD/microsoft-identity-web/issues/1097